src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java

```diff
@@ -410,11 +426,27 @@ public boolean putFileTo(String fileName, String serviceUrl) {
         return putFileToImpl(fileName, serviceUrl);
     }
 
+    /**
+     * Sends a file by HTTP PUT body to service endpoint.
+     *
+     * @param fileName   fileName to post
+     * @param partName   partName for file
+     * @param serviceUrl service endpoint to send body to.
+     * @return true if call could be made and response did not indicate error.
+     */
+    public boolean putFileAsTo(String fileName, String partName, String serviceUrl) {
+        return putFileAsToImpl(fileName, partName, serviceUrl);
+    }
+
     protected boolean putFileToImpl(String fileName, String serviceUrl) {
-        return sendFileImpl(fileName, serviceUrl, "PUT");
+        return sendFileImpl("file", fileName, serviceUrl, "PUT");
+    }
+
+    protected boolean putFileAsToImpl(String fileName, String partName, String serviceUrl) {
+        return sendFileImpl(partName, fileName, serviceUrl, "PUT");
     }
 
-    protected boolean sendFileImpl(String fileName, String serviceUrl, String method) {
+    protected boolean sendFileImpl(String partName, String fileName, String serviceUrl, String method) {
```

Comment:
Please make the order of the parameters in this method and the `...FileAsToImpl` methods the same, otherwise its too easy to make mistakes. Partname first seems most logical to me.